ArduPlane/GCS_Mavlink.cpp

```diff
@@ -794,13 +794,169 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
+                break;
+            }
+            case MAV_FRAME_GLOBAL: {
+                if   (fabsf(plane.guided_state.target_alt - new_target_alt)<0.001f ) {  // compare two floats as near-enough
+                    // reject duplicate altitude change requests
+                    return MAV_RESULT_TEMPORARILY_REJECTED;
+                }
+                plane.guided_state.target_alt = packet.z * 100;
+                break;
+            }
+            default:
+                // this wasn't a mission_item, so no forms of frame nacks are supported, MAV_RESULT_UNSUPPORED is the best we can do
+                return MAV_RESULT_UNSUPPORTED;
+        }
+
+        plane.guided_state.target_alt_frame = packet.frame;
+        plane.guided_state.last_target_alt = plane.current_loc.alt; // FIXME: Reference frame is not corrected for here
+        plane.guided_state.target_alt_time_ms = AP_HAL::millis();
+
+        if (is_zero(packet.param3)) {
+            // the user wanted /maximum acceleration, pick a large value as close enough
+            plane.guided_state.target_alt_accel = 1000.0;
+        } else {
+            plane.guided_state.target_alt_accel = fabsf(packet.param3);
+        }
+
+         // assign an acceleration direction
+        if (plane.guided_state.target_alt < plane.current_loc.alt) {
+            plane.guided_state.target_alt_accel *= -1.0f;
+        }
+        return MAV_RESULT_ACCEPTED;
+    }
+
+     case MAV_CMD_GUIDED_CHANGE_HEADING: {
+
+        // command is only valid in guided mode
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // don't accept packets outside of [0-360] degree range
+        if (packet.param2 < 0.0f || packet.param2 >= 360.0f) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        float new_target_heading = radians(wrap_180(packet.param2));
+
+        // reject duplicate heading requests if our target-heading has been set to anything except it's default value of -1
+        if (( plane.guided_state.target_heading > -0.001f ) && (fabsf(new_target_heading - plane.guided_state.target_heading)<0.001f)) { // compare two floats as near-enough
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        // course over ground
+        if ( int(packet.param1) == HEADING_TYPE_COURSE_OVER_GROUND) { // compare as nearest int
+            plane.guided_state.target_heading_type = GUIDED_HEADING_COG;
+            plane.prev_WP_loc = plane.current_loc;
+        // normal vehicle heading
+        } else if (int(packet.param1) == HEADING_TYPE_HEADING) { // compare as nearest int
+            plane.guided_state.target_heading_type = GUIDED_HEADING_HEADING;
+        } else {
+            // unknown heading track type
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        plane.g2.guidedHeading.reset_I();
```

Comment:
instead of rejecting small changes, if the issue is the I reset then just avoid the I reset when change is smaller than a threshold (could be quite a large threshold I think, maybe 5 degrees)

Reply:
it's not really about "rejecting small changes" as it is about rejecting outright duplicates.

Reply:
i have resolved this by accepting duplicates without any 'rejection' flag, as suggesed in a voice chat